### DISCUSSION TITLE
cloudflare detection

### DISCUSSION THREAD

**Discussion by eagle2049** [FlareSolverr/FlareSolverr#866] [General]
unable to bypass indeed, gives timeout error (waiting for just a moment title). Hosted Flaresolverr in cloud

**Comment by ilike2burnthing** [comment]
Make sure you're using the latest version - https://github.com/FlareSolverr/FlareSolverr/releases

Follow the troubleshooting steps here - https://github.com/FlareSolverr/FlareSolverr/wiki/Troubleshooting

If you still have problems, open a new issue and provide the information requested.

**Reply by eagle2049** [reply]
I was using v3.2.0 without proxy deployed in docker container and all requests were getting timed out. Then I upgraded to latest version but still no improvement. However I'm requests are successful in my local machine but it's failing in production.

**Reply by ilike2burnthing** [reply]
You've followed the troubleshooting steps linked above?

**Reply by ilike2burnthing** [reply]
Assuming that's your second account, access FlareSolverr in browser and copy and paste the response.

**Reply by eagle2049** [reply]
how to access flaresolverr in browser

**Reply by ilike2burnthing** [reply]
http://local_ip:port e.g. `http://192.168.0.123:8191`

**Reply by eagle2049** [reply]
ok you mean to run flaresolverr in my local environment and get the response? It will work give me a minute I will send the response

**Reply by eagle2049** [reply]
[response.text.zip](https://github.com/FlareSolverr/FlareSolverr/files/12366665/response.text.zip)
I have attached the response and the request is made is :
curl -L -X POST 'http://localhost:8191/v1' \
-H 'Content-Type: application/json' \
--data-raw '{
  "cmd": "request.get",
  "url":"https://nl.indeed.com/jobs?q=Industrieel+ingenieur&fromage=35&limit=50&start=0&sort=date&filter=0",
  "maxTimeout": 60000
}'

**Comment by ilike2burnthing** [comment]
As your issue is with the Docker container, you'd need to run that and then access it in browser as described above.

**Reply by eagle2049** [reply]
I hit the same request where flaresolverr is running in docker container.
this is the response:
{"status": "error", "message": "Error: Error solving the challenge. Timeout after 60.0 seconds.", "startTimestamp": 1692258124884, "endTimestamp": 1692258185870, "version": "3.3.2"}

this is the log message:
Incoming request => POST /v1 body: {'cmd': 'request.get', 'url': 'https://nl.indeed.com/jobs?q=Industrieel+ingenieur&fromage=35&limit=50&start=0&sort=date&filter=0', 'maxTimeout': 60000}
2023-08-17 07:42:06 INFO     Challenge detected. Title found: Just a moment...
Error: Error solving the challenge. Timeout after 60.0 seconds.
2023-08-17 07:43:05 INFO     Response in 60.986 s

**Reply by ilike2burnthing** [reply]
Increase the timeout to `1200000`

**Reply by eagle2049** [reply]
even with increased time out results are same

**Reply by ilike2burnthing** [reply]
Do you get the same for https://idope.se/ or https://torrentqq270.com/ ?

**Reply by eagle2049** [reply]
yes I'm getting same

**Reply by ilike2burnthing** [reply]
The FlareSolverr container is on the host network?

**Reply by ilike2burnthing** [reply]
Is Docker running on some VPS with a commercial datacentre IP address?

**Reply by eagle2049** [reply]
we are running it on aws cloud using docker compose

**Reply by eagle2049** [reply]
I have observed it when we find "just a moment" title click very is executing in loop, these are the logs. finally it times out

Challenge detected. Title found: Just a moment...
2023-08-17 12:58:28 INFO     Try to find the Cloudflare verify checkbox...
2023-08-17 12:58:29 INFO     Cloudflare verify checkbox found and clicked!
2023-08-17 12:58:41 INFO     Try to find the Cloudflare verify checkbox...
2023-08-17 12:58:42 INFO     Cloudflare verify checkbox found and clicked!
2023-08-17 12:58:54 INFO     Try to find the Cloudflare verify checkbox...
2023-08-17 12:58:55 INFO     Cloudflare verify checkbox found and clicked!
2023-08-17 12:59:07 INFO     Try to find the Cloudflare verify checkbox...
2023-08-17 12:59:07 INFO     Cloudflare verify checkbox found and clicked!
2023-08-17 12:59:18 ERROR    Error: Error solving the challenge. Timeout after 60.0 seconds.
2023-08-17 12:59:18 INFO     Response in 60.84 s

**Reply by ilike2burnthing** [reply]
Could be that the IP is deemed suspicious. You could try using your home IP as a proxy and make the request from your home IP as well.

**Reply by eagle2049** [reply]
we tried in different aws region as well, it didn't work

**Reply by ilike2burnthing** [reply]
Given that I know it works on Docker, and nothing else you've described about your setup seems unusual, I think AWS is the issue.

**Reply by eagle2049** [reply]
ok, thanks for you time @ilike2burnthing

**Comment by eagle2049** [comment]
I took screenshot before and after click verify, cloudflare is giving click verify challenge multiple times for the same request. I have attached the screenshots of them.
![checkbox_clicked_1_1](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/5b676fa0-c63f-462d-b81a-3b37b8e827a9)
![checkbox_clicked_1_2](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/d4450043-bc3f-43cd-b6f6-75bbea6284a1)
![checkbox_clicked_1_3](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/4da66915-c354-4361-b466-f3f88d5efbbd)
![checkbox_clicked_1_4](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/c50ebfe7-7aff-42f7-945c-146844faa698)
![checkbox_clicked_2_1](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/f375939c-b31b-4df3-af28-960b06a22fb3)
![checkbox_clicked_2_2](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/ddf60aca-4b19-4621-9e30-7f3ea8cb0b2e)
![checkbox_clicked_2_3](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/581704c2-0ca5-40e0-b879-58a08fd85430)
![checkbox_clicked_2_4](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/e52083c4-e57b-411e-a6a9-abf79eb85119)
![checkbox_clicked_3_1](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/508cf109-78bd-4995-bce7-5fd58af3788b)
![checkbox_clicked_3_2](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/cc5107d4-3d20-4d6b-b223-222f082c969e)
![checkbox_clicked_3_3](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/4b9f87b2-3f8c-4cef-8b13-3d5ef90c7d2b)
![checkbox_clicked_3_4](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/81dbb509-b41a-41f5-a72f-32c03477a8ee)
![checkbox_clicked_4_1](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/cb88b2cb-13ad-4e34-8906-663a02e8a610)
![checkbox_clicked_4_2](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/76fab14a-dd81-4e6e-bc66-882fda56fdca)
![checkbox_clicked_4_3](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/f7ecc989-effd-48e1-9864-5a3f615badf5)
![checkbox_clicked_4_4](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/7141331a-a571-4e86-ad26-0c0938563635)
![checkbox_clicked_5_1](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/ead3857e-5344-4ca4-8e63-886743bacdce)

**Comment by eagle2049** [comment]
Hi @ilike2burnthing  past one month I changed my aws region and I was able to scrape "indeed". But from last two days I'm stuck with different problem now. Right now challenge is getting detected and also solved but response html has only "Forbidden" text with 200 status code. Is is because it's unable to bypass cloudflare or "indeed" is giving trouble. 
This is the response html. I have also attached the screenshots.

<head><meta name=\"color-scheme\" content=\"light dark\"/></head><body><pre style=\"word-wrap: break-word; white-space: pre-wrap;\">Forbidden</pre></body>

![checkbox_clicked_2_4](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/a9e64bb2-4a23-4dd5-a17e-fb779ff20aea)
![checkbox_clicked_3_5](https://github.com/FlareSolverr/FlareSolverr/assets/142395694/18cf3ec2-4d01-4e4d-8087-83f657d64e56)

**Reply by ilike2burnthing** [reply]
That's the actual site returning `Forbidden`. Either it's a site issue or they've flagged the IP.

**Reply by eagle2049** [reply]
okay

**Comment by ProtDos** [comment]
I seem to have a problem accessing: https://getalby.com/auth/users/new. My error is: `{"status": "error", "message": "Error: Error solving the challenge. Timeout after 60.0 seconds.", "startTimestamp": 1698256451355, "endTimestamp": 1698256514606, "version": "3.3.6"}
`

**Reply by ilike2burnthing** [reply]
Follow the troubleshooting steps here - https://github.com/FlareSolverr/FlareSolverr/wiki/Troubleshooting

If you still have issues, open a new issue and provide the information requested.
